src/rulesets/Base/sort.jl

```diff
@@ -0,0 +1,40 @@
+function rrule(::typeof(partialsort), xs::AbstractVector, k::Union{Integer,OrdinalRange}; kwargs...)
+    inds = partialsortperm(xs, k; kwargs...)
+    ys = xs[inds]
+
+    function partialsort_pullback(Δys)
+        function partialsort_add!(Δxs)
+            for (Δy, i) in zip(Δys, inds)
+                Δxs[i] += Δy
+            end
```

Comment:
nice catch, done. without the dot though, because
```julia
julia> Δxs[4] .+= 2
ERROR: MethodError: no method matching copyto!(::Float64, ::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{0}, Tuple{}, typeof(+), Tuple{Float64, Int64}})
Closest candidates are:
  copyto!(::Any, ::Base.Broadcast.Broadcasted{var"#s155", Axes, F, Args} where {var"#s155"<:StaticArrays.StaticArrayStyle, Axes, F, Args<:Tuple}) at /Users/mzgubic/.julia/packages/StaticArrays/rdb0l/src/broadcast.jl:29
  copyto!(::AbstractArray, ::Base.Broadcast.Broadcasted{var"#s832", Axes, F, Args} where {var"#s832"<:Base.Broadcast.AbstractArrayStyle{0}, Axes, F, Args<:Tuple}) at broadcast.jl:939
  copyto!(::AbstractArray, ::Base.Broadcast.Broadcasted) at broadcast.jl:936
  ...
```